tests/integration/test_metadata_model.py

```diff
@@ -0,0 +1,33 @@
+import logging
+from contextlib import contextmanager
+from unittest.mock import patch
+
+from schematic.models.metadata import MetadataModel
+from tests.conftest import metadata_model
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+
+@contextmanager
+def does_not_raise():
+    yield
+
+
+class TestMetadataModel:
+    def test_submit_filebased_manifest(self, helpers):
+        meta_data_model = metadata_model(helpers, "class_label")
+
+        manifest_path = helpers.get_data_path(
+            "mock_manifests/filepath_submission_test_manifest.csv"
+        )
+
+        with does_not_raise():
+            manifest_id = meta_data_model.submit_metadata_manifest(
+                manifest_path=manifest_path,
+                dataset_id="syn62276880",
+                manifest_record_type="file_only",
+                restrict_rules=False,
+                file_annotations_upload=True,
+            )
+            assert manifest_id == "syn62280543"
```

Comment:
Is there other items that we can assert on here? I see that it only have a single return, but perhaps you might also want to assert that some nested functions were called for the specific logic you are testing.

Here is an example where I am spying on a function I expect to be called:
https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/tests/integration/synapseclient/core/test_download.py#L27-L73

From there you can use all the usual suspects like `assert_called_once`, `assert_called_once_with`, ect...

Reply:
@BryanFauble thanks, I added on to the tests and in the process realized that there was another issue going on that's present on `develop`, so doing the extra checks were really helpful in catching that as well